tests/service/ParserTests.fs

```diff
@@ -31,16 +30,50 @@ type U2 =
     | C
 
 let x = ()
-    """
-        let (|UnionWithCases|_|) typeDefn =
-            match typeDefn with
-            | SynTypeDefn (typeRepr = SynTypeDefnRepr.Simple (SynTypeDefnSimpleRepr.Union (unionCases = cases), _)) ->
-                cases |> List.map (fun (SynUnionCase (ident = ident)) -> ident.idText) |> Some
-            | _ -> None
-
-        let (SynModuleOrNamespace (decls = decls)) = getSingleModuleLikeDecl parseResults
-        match decls with
-        | [ SynModuleDecl.Types ([ UnionWithCases ["A"]], _)
-            SynModuleDecl.Types ([ UnionWithCases ["B"; "C"] ], _)
-            SynModuleDecl.Let _ ] -> ()
-        | _ -> failwith "Unexpected tree"
+"""
+    let (|UnionWithCases|_|) typeDefn =
+        match typeDefn with
+        | SynTypeDefn (typeRepr = SynTypeDefnRepr.Simple (SynTypeDefnSimpleRepr.Union (unionCases = cases), _)) ->
+            cases |> List.map (fun (SynUnionCase (ident = ident)) -> ident.idText) |> Some
+        | _ -> None
+
+    match getSingleModuleMemberDecls parseResults with
+    | [ SynModuleDecl.Types ([ UnionWithCases ["A"]], _)
+        SynModuleDecl.Types ([ UnionWithCases ["B"; "C"] ], _)
+        SynModuleDecl.Let _ ] -> ()
+    | _ -> failwith "Unexpected tree"
+
+
+[<Test>]
+let ``Match clause 01`` () =
+    let parseResults = getParseResults """
+match () with
+| x
+"""
+    match getSingleExprInModule parseResults with
+    | SynExpr.Match (_, _, [ SynMatchClause (_, _, _, SynExpr.ArbitraryAfterError _, _, _) ], _) -> ()
+    | _ -> failwith "Unexpected tree"
+
+
+[<Test>]
+let ``Match clause 02 - When`` () =
+    let parseResults = getParseResults """
+match () with
+| x when true
+"""
+
+    match getSingleExprInModule parseResults with
+    | SynExpr.Match (_, _, [ SynMatchClause (_, _, _, SynExpr.ArbitraryAfterError _, _, _) ], _) -> ()
+    | _ -> failwith "Unexpected tree"
+
+[<Test>]
+let ``Match clause 03 - When`` () =
+    let parseResults = getParseResults """
+match () with
```

Comment:
Could you add a test for this, assuming it works?
```fsharp
match () with
| x 
| _ -> ()
```

Reply:
Also see if this recovers:
```fsharp
match () with
| x ->
| _ -> ()
```

Reply:
Added both cases, both are green. 🙂